components/post_view/index.js

```diff
@@ -3,21 +3,48 @@
 
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
+import {withRouter} from 'react-router-dom';
+
 import {getPosts, getPostsAfter, getPostsBefore, getPostThread} from 'mattermost-redux/actions/posts';
 import {getChannel} from 'mattermost-redux/selectors/entities/channels';
 import {makeGetPostsAroundPost, makeGetPostsInChannel} from 'mattermost-redux/selectors/entities/posts';
-import {get} from 'mattermost-redux/selectors/entities/preferences';
 import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {getTeamByName} from 'mattermost-redux/selectors/entities/teams';
 
 import {increasePostVisibility} from 'actions/post_actions.jsx';
 import {checkAndSetMobileView} from 'actions/views/channel';
-import {Preferences} from 'utils/constants.jsx';
+import {makePreparePostIdsForPostList} from 'selectors/posts';
+import {Constants} from 'utils/constants.jsx';
 
 import PostList from './post_list.jsx';
 
+// This function is added as a fail safe for the channel sync issue we have.
+// When the user switches to a team for the first time we show the channel of previous team and then settle for the right channel after that
+// This causes the scroll correction etc an issue because post_list is not mounted for new channel instead it is updated
+const isChannelLoading = (params, channel, team) => {
+    if (params.postid) {
+        return false;
+    }
+    if (channel && team) {
+        if (channel.type !== Constants.DM_CHANNEL && channel.type !== Constants.GM_CHANNEL) {
+            if (channel.name !== params.identifier) {
+                return true;
+            }
+        }
+        if (channel.team_id && channel.team_id !== team.id) {
+            return true;
+        }
+    } else {
+        return true;
+    }
+
+    return false;
+};
```

Comment:
Non-blocking but the last part is a bit confusing to read (at least for me).  Could it be simplified?  Any chance that it's same as,
```javascript
const isChannelLoading = (params, channel, team) => {
    if (params.postid) {
        return false;
    }

    if (channel && team) {
        if (channel.type !== Constants.DM_CHANNEL && channel.type !== Constants.GM_CHANNEL) {
            if (channel.name !== params.identifier) {
                return true;
            }
        }
        if (channel.team_id && channel.team_id !== team.id) {
            return true;
        }

        return false;
    }

    return true;
};
```

Reply:
Yup, Makes sense. Fixed it